orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationAiModel.java

```diff
@@ -114,4 +117,58 @@ public class OrchestrationAiModel {
   LLMModuleConfig createConfig() {
     return LLMModuleConfig.create().modelName(name).modelParams(params).modelVersion(version);
   }
+
+  /**
+   * Additional parameter on this model.
+   *
+   * @param key the parameter key.
+   * @param value the parameter value, nullable.
+   * @return A new model with the additional parameter.
+   */
+  @Nonnull
+  public OrchestrationAiModel withParam(@Nonnull final String key, @Nullable final Object value) {
+    final var params = new LinkedHashMap<>(getParams());
+    params.put(key, value);
+    return withParams(params);
+  }
+
+  /**
+   * Additional parameter on this model.
+   *
+   * @param param the parameter key.
+   * @param value the parameter value, nullable.
+   * @return A new model with the additional parameter.
+   */
+  @Nonnull
+  public OrchestrationAiModel withParam(
+      @Nonnull final Parameter param, @Nullable final Object value) {
+    return withParam(param.value, value);
+  }
+
+  /** Parameter key for a model. */
+  @RequiredArgsConstructor
+  public enum Parameter {
+    /** The maximum number of tokens to generate. */
+    MAX_TOKENS("max_tokens"),
+    /** The sampling temperature. */
+    TEMPERATURE("temperature"),
+    /** The frequency penalty. */
+    FREQUENCY_PENALTY("frequency_penalty"),
+    /** The presence penalty. */
+    PRESENCE_PENALTY("presence_penalty"),
+    /** The maximum number of tokens for completion */
+    MAX_COMPLETION_TOKENS("max_completion_tokens"),
+    /** The probability mass to be considered . */
+    TOP_P("top_p"),
+    /** The toggle to enable partial message delta. */
+    STREAM("stream"),
+    /** The options for streaming response. */
+    STREAM_OPTIONS("stream_options"),
+    /** The tokens where the API will stop generating further tokens. */
+    STOP("stop"),
```

Comment:
Isn't this the same as `max_tokens` then?

Reply:
If I understood correctly, it refers to a stop token / literal / string, on which it will stop generating - or on which it signals generation stop.